src/flask_app.py

```diff
@@ -50,5 +46,5 @@ def flask_environment(self) -> dict[str, str]:
         }
         secret_key_env = f"{FLASK_ENV_CONFIG_PREFIX}SECRET_KEY"
         if secret_key_env not in env:
-            env[secret_key_env] = self._secret_storage.get_flask_secret_key()
+            env[secret_key_env] = self._charm_state.secret_storage.get_flask_secret_key()
```

Comment:
Should this be abstracted, e.g., `CharmState` has a `secret_key` property which calls `secret_storage.get_flask_secret_key()` since otherwise the `CharmState` needs to have a mock implementation of `SecretStorage` for testing

Reply:
The ops testing harness will automatically provide simulation for peer relation, so we don't need to mock the `SecretStorage` in unit tests ourself, just `harness.begin_with_initial_hooks()` is enough.

Reply:
The thing is that the aim is to be able to test modules without needing the harness, calling the function means that any modules that depend on the secret store will require to have unit tests with the harness

Reply:
I agree, and have a similar comment for L134 and L230 in charm.py
I think that the charmstate shouldn't store the whole SecretStorage, but just the `get_flask_secret_key()` method
Calling a secretstorage method through the charmstate seems strange to me

What do you think @jdkandersson .

Reply:
The `SecretStorage` doesn't just store the secret key, there's also a method called `is_initialized` to check if the `SecretStorage` is initialized. Read or write the `SecretStorage` is an error before initialization.